tests/test_sriov_device_plugin.py

```diff
@@ -0,0 +1,52 @@
+import os
+import pytest
+import platform
+
+from subprocess import PIPE, STDOUT, CalledProcessError, run
+from utils import microk8s_disable
+
+
+class TestSRIOVDevicePlugin(object):
+    """SR-IOV Network Device Plugin relies on availability of given PCI devices, so we can only
+    test for the exception being raised.
+    """
+
+    @pytest.mark.skipif(
+        platform.machine() != "x86_64",
+        reason="SR-IOV Network Device Plugin tests are only relevant in x86 architectures",
+    )
+    @pytest.mark.skipif(
+        os.environ.get("STRICT") == "yes",
+        reason="Skipping sriov-device-plugin tests in strict confinement as they are expected to fail",  # noqa: E501
+    )
+    def test_sriov_device_plugin(self):
+        """
+        Sets up and validates SR-IOV Network Device Plugin.
+        """
+        script_path = os.path.abspath(os.path.dirname(__file__))
+        sriov_resources_mapping_file_name = "sriov-device-plugin-test-resources.json"
+        sriov_resources_mapping_file = os.path.join(
+            script_path,
+            "resources",
+            sriov_resources_mapping_file_name,
+        )
+        enable_sriov_dp_cmd = [
+            "/snap/bin/microk8s.enable",
+            "sriov-device-plugin",
+            "--resources-file",
+            sriov_resources_mapping_file,
+        ]
+        print("Enabling SR-IOV Network Device Plugin")
+        try:
+            cmd = run(
+                enable_sriov_dp_cmd,
+                stdout=PIPE,
+                input=b"N\n",
+                stderr=STDOUT,
+                check=True,
+            )
+            assert "ValueError: Invalid device PCI address!" in cmd.stderr.decode()
```

Comment:
I wonder if we can test to ensure everything works as expected (instead of testing only that the exception is raised). That'd probably require patching the `_pci_address_is_valid` method.

Reply:
+1 Maybe check for the sriov test plugin resources.

Reply:
yea I'm working on this (currently on the k8s-snap integration tests tho)

Reply:
hey guys, need a bit of help with elementary school python. was trying to write a `happy scenario` test but I'm not able to mock necessary functions. seems like [it's not possible to globally mock something](https://docs.python.org/dev/library/unittest.mock.html#where-to-patch). also, we don't have access to `addons.sriov_device_plugin.enable` because the file is not named `enable.py` (and can't be changed IMO, since it'll break stuff). any ideas? @louiseschmidtgen @mateoflorido